Userland/Libraries/LibGUI/FileSystemModel.cpp

```diff
@@ -700,8 +701,18 @@ static Threading::MutexProtected<ThumbnailCache> s_thumbnail_cache {};
 
 static ErrorOr<NonnullRefPtr<Gfx::Bitmap>> render_thumbnail(StringView path)
 {
-    Gfx::IntSize thumbnail_size { 32, 32 };
-    auto bitmap = TRY(Gfx::Bitmap::load_from_file(path, 1, thumbnail_size));
+    Core::EventLoop event_loop;
+
+    auto file = TRY(Core::MappedFile::map(path));
+    auto client = TRY(ImageDecoderClient::Client::try_create());
+    auto mime_type = Core::guess_mime_type_based_on_filename(path);
+    auto decoded_image = client->decode_image(file->bytes(), mime_type);
```

Comment:
This no longer passes the 'ideal_size' to the decoder, so will result in scaling artifacts for vector graphics.